File: Leandro-Gonzalez/Java/Clase12/Ventas/src/codesystem/ventas/Orden.java
package codesystem.ventas;

public class Orden {
    // Atributos de la clase
    private int idOrden;
    private Producto productos[]; // Arreglo de objetos declarado
    private static int contadorOrdenes;
    private int contadorProductos;
    private static final int MAX_PRODUCTOS = 10;

    // Constructor vacio
    public Orden() {
        this.idOrden = ++Orden.contadorOrdenes; // Incrementa el contador de ordenes y asigna el valor al idOrden
        this.productos = new Producto[Orden.MAX_PRODUCTOS]; // Inicializa el arreglo de productos
    }

    // Metodo que combina la clase Producto con la clase Orden
    public void agregarProducto(Producto producto) {
        // Verifica si el contador de productos es menor que el maximo de productos
        if (this.contadorProductos < Orden.MAX_PRODUCTOS) {
            // Agrega el producto al arreglo e incrementa el contador
            this.productos[this.contadorProductos++] = producto;
        } else {
            System.out.println("Se ha superado el maximo de productos: " + Orden.MAX_PRODUCTOS);
        }
    }

    // Metodo para calcular el total de la orden
    public double calcularTotal() {
        double total = 0; // Inicializa el total en 0
        for (int i = 0; i < this.contadorProductos; i++) {
            total += this.productos[i].getPrecio(); // Suma el precio de cada producto al total
        }
        return total;
    }

    // Metodo para mostrar los datos de la orden
    public void mostrarOrden() {
        System.out.println("Orden #: " + this.idOrden);
        System.out.println("Total de la orden: $" + this.calcularTotal());
        System.out.println("Productos de la orden: ");
        for (int i = 0; i < this.contadorProductos; i++) {
            System.out.println(this.productos[i]); // Muestra los productos de la orden
        }
    }

}

File: Ignacio Villarraza/Python/Clase 11/Orden.py
from Producto import Producto #Importamos la clase Producto

#Creamos la clase Orden
class Orden:
    contador_ordenes = 0 #Variable de clase

    #Constructor de la clase
    def __init__(self, productos):
        Orden.contador_ordenes += 1 #Incremento del contador de ordenes
        self.id_orden = Orden.contador_ordenes #Asignación del id de la orden
        self._productos = list(productos) #Asignación de la lista de productos
        
        
    #Método para agregar productos a la orden
    def agregar_producto(self, producto):
        self._productos.append(producto) #Agregamos el producto a la lista de productos
        
    #Método para calcular el total de la orden
    def calcular_total(self):
        total = 0 #Variable temporal para almacenar el total de la orden
        for producto in self._productos: #Iteración sobre la lista de productos
            total += producto.precio #Suma del precio de los productos
        return total

    #Método para sobreescribir el método str
    def __str__(self):
        productos_str = '' #Variable temporal para almacenar los productos en formato string
        for producto in self._productos: #Iteramos sobre la lista de productos
            productos_str += producto.__str__() + ' | ' #Concatenamos los productos en formato string
        return f'Orden: {self.id_orden}, \nProductos: {productos_str}' #Retornamos la orden en formato string
    
#Método main solo es visible en este archivo
if __name__ == "__main__":
    producto1 = Producto("Camisa", 100.00)
    producto2 = Producto("Pantalon", 150.00)
    productos1 = [producto1, producto2] #Creamos una lista de productos
    orden1 = Orden(productos1) #Creamos una orden
    print(orden1) #Imprimimos la orden
    
    #Creamos una nueva orden con nuevos productos
    producto3 = Producto("Zapato", 200.00)
    producto4 = Producto("Gorra", 50.00)
    productos2 = [producto3, producto4] #Creamos una nueva lista de productos
    orden2 = Orden(productos2) #Creamos otra orden
    print(orden2) #Imprimimos la orden



